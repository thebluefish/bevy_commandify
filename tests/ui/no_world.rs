use bevy_commandify::*;
use bevy::prelude::*;
use bevy::ecs::system::CommandQueue;

#[command(no_world)]
fn foo(world: &mut World) { }

/// Test that our generated trait works for Commands, but not World
fn main() {
    let mut world = World::new();

    let mut queue = CommandQueue::default();
    let mut commands = Commands::new(&mut queue, &mut world);

    // success
    commands.foo();
    // failure
    world.foo();

    queue.apply(&mut world);
}