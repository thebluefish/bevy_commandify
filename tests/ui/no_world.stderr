error[E0599]: no method named `foo` found for struct `bevy::prelude::World` in the current scope
  --> tests/ui/no_world.rs:18:11
   |
18 |     world.foo();
   |           ^^^ method not found in `World`
   |
   = help: items from traits can only be used if the trait is implemented and in scope
note: `CommandsFooExt` defines an item `foo`, perhaps you need to implement it
  --> tests/ui/no_world.rs:5:1
   |
5  | #[command(no_world)]
   | ^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in the attribute macro `command` (in Nightly builds, run with -Z macro-backtrace for more info)
